compass/landice/tests/ismip6_forcing/atmosphere/create_mapfile_smb.py

```diff
@@ -0,0 +1,165 @@
+import os
+import shutil
+import subprocess
+import netCDF4
+import xarray as xr
+from mpas_tools.scrip.from_mpas import scrip_from_mpas
+from mpas_tools.logging import check_call
+
+
+# function that creates a mapping file from ismip6 grid to mali mesh
+def build_mapping_file(config, cores, logger, ismip6_grid_file,
+                       mapping_file, mali_mesh_file=None,
+                       method_remap=None):
+    """
+    Build a mapping file if it does not exist.
+
+    Parameters
+    ----------
+    config : compass.config.CompassConfigParser
+        Configuration options for a ismip6 forcing test case
+    cores : int
+        the number of cores for the ESMF_RegridWeightGen
+    logger : logging.Logger
+        A logger for output from the step
+    ismip6_grid_file : str
+        ismip6 grid file
+    mapping_file : str
+        weights for interpolation from ismip6_grid_file to mali_mesh_file
+    mali_mesh_file : str, optional
+        The MALI mesh file if mapping file does not exist
+    method_remap : str, optional
+        Remapping method used in building a mapping file
+    """
+
+    if mali_mesh_file is None:
+        raise ValueError("Mapping file does not exist. To build one, Mali "
+                         "mesh file with '-f' should be provided. "
+                         "Type --help for info")
+
+    ismip6_scripfile = "temp_ismip6_8km_scrip.nc"
+    mali_scripfile = "temp_mali_scrip.nc"
+    ismip6_projection = "ais-bedmap2"
+
+    # create the ismip6 scripfile if mapping file does not exist
+    # this is the projection of ismip6 data for Antarctica
+    print("Mapping file does not exist. Building one based on the "
+          "input/ouptut meshes")
+    print("Creating temporary scripfiles for ismip6 grid and mali mesh...")
+
+    # create a scripfile for the atmosphere forcing data
+    create_atm_scrip(ismip6_grid_file, ismip6_scripfile)
+
+    # create a MALI mesh scripfile if mapping file does not exist
+    # make sure the mali mesh file uses the longitude convention of [0 2pi]
+    # make changes on a duplicated file to avoid making changes to the
+    # original mesh file
+
+    mali_mesh_copy = f"{mali_mesh_file}_copy"
+    shutil.copy(mali_mesh_file, f"{mali_mesh_file}_copy")
+
+    args = ["set_lat_lon_fields_in_planar_grid.py",
+            "--file", mali_mesh_copy,
+            "--proj", ismip6_projection]
+
+    subprocess.check_call(args)
```

Comment:
Here and for other calls like it, this needs to use `mpas_tools.logging.check_call` and pass `self.logger`.  Otherwise, output goes to the terminal instead of the log file.